README.md

```diff
@@ -39,58 +43,615 @@ Use a `<script>` tag to reference the package. Using this method will register a
+
+#### 2. When using a package manager
 
-#### The destination
+Etvas Automat is ready to be used with `npm` or `yarn`. The installation is straightforward:
 
-As this field is available for all our display blocks, it is worth to note what you can specify for the `destination` parameter.
+```
+$ npm install @etvas/etvas-automat`
+```
 
-You can specify an ID of a HTML element which is present in your DOM:
+or, if you prefer yarn:
 
 ```
-etvasAutomat.showProductCard('#my-placeholder-div', options)
+$ yarn add @etvas/etvas-automat
 ```
 
-Also, you can directly specify the JS DOM element:
+You can use the package as you normally use any other package. If you are using the CommonJS syntax, you can require the package like this:
 
 ```
-const el = document.getElementById('my-placeholder-div')
-etvasAutomat.showProductCard(el, options)
+const etvasAutomat = require("@etvas/etvas-automat");
+
+etvasAutomat.initialize({ ... });
+```
+
+But, as this is a Front-End SDK, you might use the `import` syntax:
+
+```
+import automat from "@etvas/etvas-automat";
+
+automat.initialize({ ... });
 ```
 
-Although not recommended, if you need to display it in multiple div's, identified by a common class, you can use:
+## Initializing the library
+
+The library needs some configuration in order to work and communicate properly with our backend services. We do that by calling a function named `initialize`. You will want to call this function as early as possible in your code, making sure it is the first one called:
+
+```
+automat.initialize({
+  etvasURL: "https://appname.etvasapps.com",
+  etvasApiKey: "kn0-providedapikeystring",
+  organizationId: "12345678-1234-1234-4123-123456789012",
+  locale: "en"
+})
+```
+
+Only the `etvasURL` and the `organizationId` are required. The other ones have either a default value or they are needed just in some cases. Aside the `locale` entry, everything else is provided by Etvas Support Team or you can find the values in [Etvas Partners Portal](https://partners.helloetvas.com)
+
+#### `etvasURL`
+
+This is the URL you can find in your account on [Etvas Partner's Studio](https://partners.helloetvas.com). We call this URL the Customer Portal, the portal tailored and branded by you to show it to your customers. It is required in order to preserve the brand colors and graphics you provided, so the integration in your application is as natural as possible.
+
+#### `etvasApiKey`
+
+This is a public key we provide (and you can find it in your account in [Etvas Partner's Studio](https://partners.helloetvas.com)) which is used not only to uniquely identify your app (the public part) but also to watch, analyze and shape the incoming traffic to automatically protect against a wide range of network attacks. Please note the entire process is anonymous and no personal data is stored by these operations.
```

Comment:
is this the Partner API key or the graphql public api key ?

Reply:
Its the graphql one I think (`da2-xxx`).

Reply:
in that case I don't believe we expose it in the Partner's Studio ... 
that one is a global API Key ... not even one per partner :|

Reply:
Yes. I think this is the API key that is used by our customer portal(s) to make unauthenticated calls. Unfortunately the `getBrandingBySlug` GraphQL query falls into that category.

Being this case, I really don't want to give them this API key to them. We should discuss options.

Reply:
One option would be to: Use it internally in etvas-automat and not document that you can override it.

Reply:
Done